deployment/actionManger.go

```diff
@@ -221,63 +254,111 @@ func (mgr *ActionManager) activateMissing(toCheck worker.ComponentID) error {
 		mgr.pathHashes[worker.ComponentPath{
 			User: toCheck.User,
 			Repo: toCheck.Repo,
-		}] = activatedHash
+		}].hash = activatedHash
 	}
 
 	return nil
 }
 
-func (mgr *ActionManager) ensureSomeWorkerIsRunning(compID worker.ComponentID) error {
+func (mgr *ActionManager) ensureNWorkerIsRunning(compID worker.ComponentID) error {
```

Comment:
I need you to refactor this function more. It's unreadable and I can't process the logic as-is